contributed_definitions/nyaml/NXcoordinate_system_set.yaml

```diff
@@ -88,138 +88,259 @@ doc: |
-#          
-#          In effect, 1 should be the preferred choice. However, if more than one coordinate
-#          system is defined for practical purposes, explicit depends_on fields should
-#          always guide the user for each group and field which of the coordinate system
-#          one refers to.
-#     </doc>
-#     <group type="NXcoordinate_system"/>
-#     <!--How to interpret "depends_on" calls for NXtransformations instances which live
-# within an instance of NXcoordinate_system(_set)?
-# Depends on cardinality of NXcoordinate_system_set, i.e. how many NXcoordinate_system are defined?
-# - 0, the root of an NXtransformation chain i.e. "." means McStas
-# - 1, the root of an NXtransformation chain i.e. "." means "that specific CS" i.e. the one defined by the single instance of NXcoordinate_system in the set
-# - > 1, "." becomes ambiguous. My suggestion, spell out the name of the specific NXcoordinate_system instance to be used, using "."
-# despite all these, a warning should be raised, like a logic warning (and McStas kick in) or an error (i.e. be forbidden) be raised-->
-# </definition>
+  # frequently used coordinate systems with frequently relevant specializations
+  processing_reference_frame(NXcoordinate_system):
+    doc: |
+      Details about eventually relevant named directions that may
+      give reasons for anisotropies. The classical example is cold-rolling
+      where one has to specify which directions (rolling, transverse, and normal)
+      align how with the direction of the base vectors of the sample_reference_frame.
+    type(NX_CHAR):
+      doc: |
+        Type of coordinate system and reference frame according to
+        convention 1 of DOI: 10.1088/0965-0393/23/8/083501.
+      enumeration: [undefined, cartesian]
+    handedness(NX_CHAR):
+      doc: |
+        Handedness of coordinate system.
+      enumeration: [right_handed, left_handed]
+    origin(NX_CHAR):
+      doc: |
+        Location of the origin of the processing_reference_frame.
+        This specifies the location Xp = 0, Yp = 0, Zp = 0.
+        Assume regions-of-interest in this reference frame form a
+        rectangle or cuboid.
+        Edges are interpreted by inspecting the direction of their
+        outer unit normals (which point either parallel or antiparallel)
+        along respective base vector direction of the reference frame.
+      enumeration: [undefined, front_top_left, front_top_right, front_bottom_right, front_bottom_left, back_top_left, back_top_right, back_bottom_right, back_bottom_left]
+    x_alias(NX_CHAR):
+      doc: |
+        Name or alias assigned to the x-axis base vector,
+        e.g. rolling direction.
+    x_direction(NX_CHAR):
+      doc: |
+        Direction of the positively pointing x-axis base vector of
+        the processing_reference_frame. We assume the configuration
+        is inspected by looking towards the sample surface from a position
```

Comment:
Way is that an assumption that should always be taken? The individual communities can have different agreements about their prefered coordinate system.

Reply:
Exactly so then only possible in appdef but what if 20 appdefs agree on that aspect while showing differences on other parts of their concept tree, this is what I do not find convincing about NeXus, introducing reusable base classes that are allowed to make more domain specific conventions PROVIDED they otherwise build rigorously on core NIAC accepted base classes is in my opinion they way to go and maybe eventually even a practical way to go. I see that members in the NIAC may not feel to have the authority to decide what a specific community would like to agree on concept-wise, exactly this is exactly @yayahjb point about that also NeXus needs to face the reality of how to place oneself and be specific about what it serves it customers in a sea of options like UML, semantic web technology stack solutions and classical ad hoc XML or JSON-LD schema writing approaches that is a key question we have to ask ourselves which solution we propose.

My proposal is as I said:
1.) Core generic base classes (not a glossary of all human mankind terms)
2.) The man in the middle reusable concept sub-graphs (eventually allowing setting of constraints)
2.) Appdefs for the most specific combination of concepts + their constraints